invoke/tasks.py

```diff
@@ -357,7 +358,7 @@ def inner(body: Callable) -> Task[T]:
         return _task
 
     # update_wrapper(inner, klass)
-    return inner
+    return cast('Task["T"]', inner)
```

Comment:
Maybe this'll be fixed by the above comment, but if not, it'd be better to type ignore than cast here, given that it is literally not a Task being returned.

Reply:
This is following the recommendation provided by mypy documentation: https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators

Reply:
Right, I see what you're pointing to. It's not actually recommended, but maybe needed if we don't use typing-extensions to get ParamSpec.

Reply:
No, wait, my point was that this is not a Task. The docs you point to are accepting a function and returning a function, the cast is to the function type. This is returning a function and then you're casting it as a Task.

docs:
```
F = TypeVar('F', bound=Callable[..., Any])
def printing_decorator(func: F) -> F:
    return cast(F, wrapper)
```

here:
```
def task(*args: Any, **kwargs: Any) -> Task[T]:
    def inner(body: Callable) -> Task[T]:
        ...
    return cast(Task[T], inner)  # inner is clearly not a Task
```